modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APIAccessibilityOfPublishedOldAPIAndPublishedCopyAPITestCase.java

```diff
@@ -0,0 +1,179 @@
+        HttpResponse httpResponseCopyAPI =
+                apiPublisherClientUser1.copyAPI(API1_PROVIDER_NAME, API1_NAME, API_VERSION1, API_VERSION2, "");
+
+        assertEquals(httpResponseCopyAPI.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                "Copy API request code is invalid." + getAPIIdentifierString(apiIdentifierAPI1Version1));
+        assertEquals(getValueFromJSON(httpResponseCopyAPI, "error"), "false",
+                "Copy  API response data is invalid" + getAPIIdentifierString(apiIdentifierAPI1Version1) +
+                        "Response Data:" + httpResponseCopyAPI.getData());
+
+    }
+
+
+    @Test(groups = {"wso2.am"}, description = " Test Copy API.", dependsOnMethods = "testCopyAPI")
+    public void testPublishCopiedAPI() throws Exception {
+
+        //Publish  version 2.0.0
+        APILifeCycleStateRequest publishUpdateRequest =
+                new APILifeCycleStateRequest(API1_NAME, API1_PROVIDER_NAME, APILifeCycleState.PUBLISHED);
+        publishUpdateRequest.setVersion(API_VERSION2);
+        HttpResponse publishAPIResponse =
+                apiPublisherClientUser1.changeAPILifeCycleStatusToPublish(apiIdentifierAPI1Version2, false);
+        assertEquals(publishAPIResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                "API publish Response code is invalid " + getAPIIdentifierString(apiIdentifierAPI1Version2));
+        assertTrue(verifyAPIStatusChange(publishAPIResponse, APILifeCycleState.CREATED, APILifeCycleState.PUBLISHED),
+                "API status Change is invalid in" + getAPIIdentifierString(apiIdentifierAPI1Version2) +
+                        "Response Data:" + publishAPIResponse.getData());
+
+
+    }
+
+
+    @Test(groups = {"wso2.am"}, description = " Test availability of old and new API versions in the store.",
+            dependsOnMethods = "testPublishCopiedAPI")
+    public void testAvailabilityOfOldAndNewAPIVersionsInStore() throws Exception {
+
+        // Check availability of old API version in API Store
+        List<APIIdentifier> apiStoreAPIIdentifierList =
+                APIMgtTestUtil.getAPIIdentifierListFromHttpResponse(apiStoreClientUser1.getAPI(API1_NAME));
+        assertEquals(APIMgtTestUtil.isAPIAvailable(apiIdentifierAPI1Version1, apiStoreAPIIdentifierList), true,
+                "Old version Api is not visible in API Store after publish new version." + getAPIIdentifierString(
+                        apiIdentifierAPI1Version1));
+
+        // Check availability of new API version in API Store
+        apiStoreAPIIdentifierList =
+                APIMgtTestUtil.getAPIIdentifierListFromHttpResponse(apiStoreClientUser1.getAPI(API1_NAME));
+        assertEquals(APIMgtTestUtil.isAPIAvailable(apiIdentifierAPI1Version2, apiStoreAPIIdentifierList), true,
+                "New version Api is not visible in API Store after publish new version." + getAPIIdentifierString(
+                        apiIdentifierAPI1Version2));
+    }
+
```

Comment:
Maintain a generic and meaning full naming conversion from beginning avoid use names such as oldVersion, newVersion and version1, version2 so on . Keep naming consistant through out the class to improve readability and understandability of the code